src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.cs

```diff
@@ -92,20 +92,36 @@ public static void EmptyEntryTest(ZipArchiveMode mode)
             baseline = baseline.Clone();
             using (ZipArchive archive = new ZipArchive(baseline, mode))
             {
-                AddEntry(archive, "data1.txt", data1, lastWrite);
+                if (mode == ZipArchiveMode.Create)
+                {
+                    AddEntry(archive, "data1.txt", data1, lastWrite);
 
-                ZipArchiveEntry e = archive.CreateEntry("empty.txt");
-                e.LastWriteTime = lastWrite;
-                using (Stream s = e.Open()) { }
+                    ZipArchiveEntry e = archive.CreateEntry("empty.txt");
+                    e.LastWriteTime = lastWrite;
+                    using (Stream s = e.Open()) { }
+                }
+                else
+                {
+                    ZipArchiveEntry e = archive.GetEntry("data2.txt");
+                    e.Delete();
```

Comment:
The changes in this unit test do not seem related to the changes. There's no duplication attempt being tested.

Can you please remove the changes to this file? I think the unit test you added in the other file is doing all the verification we need.

Reply:
This test attempts to add twice the same files: data1.txt, data2.txt, and empty.txt inside the zip archive when ZipArchiveMode.Update, so the test crashes with the introduced InvalidOperationException exception. The test works fine when ZipArchiveMode.Create because ZipArchive class overwrites the existing zip files.

Reply:
Ah, I see what you mean. Before your changes, the `AddEntry` call would crash.

Still, my comment applies: the current change isn't testing your new code.

I suggest you remove your if elses, keep the same code as before, _but_ the `AddEntry` invocations inserting a duplicate entry should be wrapped in an `Assert.Throws`. That way, not only we test your new code, but we also keep going with the rest of the test code, and there shouldn't be a duplicate entry accidentally inserted in the end (the last line converting the streams to arrays and comparing them would verify this).

Agree?

Reply:
I can't remove the if/else, because the InvalidOperationException is thrown only in case of ZipArchiveMode.Update. The case with ZipArchiveMode.Create works fine.

Reply:
Sorry for my bad english, maybe I didn't explain well my point of vue. The test to pass needs to be like this :

https://github.com/pedrobsaila/runtime/blob/2887dbee99f2c211a16782adc83563281fbfa757/src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.cs#L91-L125. So I rollbacked the code and catched the exception but still letting the if/else